core/src/main/java/org/apache/iceberg/DefaultValueParser.java

```diff
@@ -0,0 +1,190 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.UUID;
+import org.apache.iceberg.expressions.Literal;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.relocated.com.google.common.io.BaseEncoding;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+import org.apache.iceberg.util.DateTimeUtil;
+
+public class DefaultValueParser {
+  private DefaultValueParser() {
+  }
+
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
+  public static Object parseDefaultFromJson(Type type, JsonNode defaultValue) {
+
+    if (defaultValue == null || defaultValue.isNull()) {
+      return null;
+    }
+
+    switch (type.typeId()) {
+      case BOOLEAN:
+        Preconditions.checkArgument(defaultValue.isBoolean(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        return defaultValue.booleanValue();
+      case INTEGER:
+        Preconditions.checkArgument(defaultValue.isIntegralNumber() && defaultValue.canConvertToInt(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        return defaultValue.intValue();
+      case LONG:
+        Preconditions.checkArgument(defaultValue.isIntegralNumber() && defaultValue.canConvertToLong(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        return defaultValue.longValue();
+      case FLOAT:
+        Preconditions.checkArgument(defaultValue.isNumber(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        return defaultValue.floatValue();
+      case DOUBLE:
+        Preconditions.checkArgument(defaultValue.isNumber(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        return defaultValue.doubleValue();
+      case DECIMAL:
+        Preconditions.checkArgument(defaultValue.isNumber(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        return defaultValue.decimalValue();
+      case STRING:
+        Preconditions.checkArgument(defaultValue.isTextual(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        return defaultValue.textValue();
+      case UUID:
+        Preconditions.checkArgument(defaultValue.isTextual(),
+            "Cannot parse %s to a %s value", defaultValue, type);
+        return UUID.fromString(defaultValue.textValue());
```

Comment:
I think this should validate that the string's length is the length of a UUID string.

Reply:
`UUID.fromString` is already doing such validation, let me use a try-catch block to print out the error message with the same format?

Reply:
Actually, it isn't:

```java
    public static UUID fromString(String var0) {
        String[] var1 = var0.split("-");
        if (var1.length != 5) {
            throw new IllegalArgumentException("Invalid UUID string: " + var0);
        } else {
            for(int var2 = 0; var2 < 5; ++var2) {
                var1[var2] = "0x" + var1[var2];
            }

            long var6 = Long.decode(var1[0]);
            var6 <<= 16;
            var6 |= Long.decode(var1[1]);
            var6 <<= 16;
            var6 |= Long.decode(var1[2]);
            long var4 = Long.decode(var1[3]);
            var4 <<= 48;
            var4 |= Long.decode(var1[4]);
            return new UUID(var6, var4);
        }
    }
```

It's just validating that there are enough parts, and decoding those parts.

Instead of doing the try/catch thing (which has its own problem) I think you should check that the string is the right length to be a UUID.